table/table.go

```diff
@@ -0,0 +1,97 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package table
+
+import (
+	"reflect"
+
+	"github.com/apache/iceberg-go"
+	"github.com/apache/iceberg-go/io"
+	"golang.org/x/exp/slices"
+)
+
+type Identifier = []string
+
+type Table struct {
+	identifier       Identifier
+	metadata         Metadata
+	metadataLocation string
+	fs               io.IO
+}
+
+func (t Table) Equals(other Table) bool {
+	return slices.Equal(t.identifier, other.identifier) &&
+		t.metadataLocation == other.metadataLocation &&
+		reflect.DeepEqual(t.metadata, other.metadata)
```

Comment:
so comparing metadata directly might be tricky due to Schema equality. You might want to look at the Java impl on how Schema comparison is performed (basically the schema ID is not taken into account, because a schema ID can be re-assigned)

Reply:
I think we can go ahead and merge this PR, but this is something to keep in mind for a follow-up